packages/plugins/apollo-usage-report/src/index.ts

```diff
@@ -55,24 +59,82 @@ type ApolloUsageReportOptions = ApolloInlineTracePluginOptions & {
    * Client version to report to the usage reporting API
    */
   clientVersion?: StringFromRequestFn;
+  /**
+   * The version of the runtime (like 'node v23.7.0')
+   * @default empty string.
+   */
+  runtimeVersion?: string;
+  /**
+   * The hostname of the machine running this server
+   * @default $HOSTNAME environment variable
+   */
+  hostname?: string;
+  /**
+   * The OS identification string.
+   * The format is `${os.platform()}, ${os.type()}, ${os.release()}, ${os.arch()})`
+   * @default empty string
+   */
+  uname?: string;
+  /**
+   * The maximum estimated size of each traces in bytes. If the estimated size is higher than this threshold,
+   * the complete trace will not be sent and will be reduced to aggregated stats.
+   *
+   * Note: GraphOS only allow for traces of 10mb maximum
+   * @default 10 * 1024 * 1024 (10mb)
+   */
+  maxTraceSize?: number;
+  /**
+   * The maximum uncompressed size of a report in bytes.
+   * The report will be sent once this threshold is reached, even if the delay between send is not
+   * yet expired.
+   *
+   * @default 4Mb
+   */
+  maxBatchUncompressedSize?: number;
+  /**
+   * The maximum time in ms between reports.
+   * @default 20s
+   */
+  maxBatchDelay?: number;
+  /**
+   * Control if traces should be always sent.
+   * If false, the traces will be batched until a delay or size is reached.
+   * Note: This is highly not recommended in a production environment
+   *
+   * @default false
+   */
+  alwaysSend?: boolean;
+  /**
+   * Timeout in ms of a trace export tentative
+   * @default 30s
+   */
+  exportTimeout?: number;
+  /**
+   * The class to be used to keep track of traces and send them to the GraphOS endpoint
+   * Note: This option is aimed to be used for testing purposes
+   */
+  reporter?: (
```

Comment:
@EmrysMyrddin Why are the `Reporter` constructor options exposed to the client? I don't see much reason for this and I don't think it's right.

Reply:
I would recommend not export this to the client side unless it is really necessary and I would rather mock those "options" via spyOn/replaceProperty

Reply:
@EmrysMyrddin I personally don't like it because I don't believe that _"users will just not use it"_. 😄  But if this is the usual pattern here, then it's fine.
Could you please release a new alpha version for me to test?

Reply:
Sorry, I already see the current (`@graphql-yoga/plugin-apollo-usage-report@0.9.0-alpha-20250602093039-da21e7b1322d204bdc7f3b2737561e1c26c2bc17`) version above. I didn't notice that the post above from `github-actions` bot was updated.